velox/functions/prestosql/JsonFunctions.cpp

```diff
@@ -84,38 +162,71 @@ class JsonParseFunction : public exec::VectorFunction {
       auto value = arg->as<ConstantVector<StringView>>()->valueAt(0);
       paddedInput_.resize(value.size() + simdjson::SIMDJSON_PADDING);
       memcpy(paddedInput_.data(), value.data(), value.size());
-      if (auto error = parse(value.size())) {
+      auto escapeSize = escapedStringSize(value.data(), value.size(), true);
```

Comment:
The escapeSize here is worst case size , Its hard to estimate the size only for the string nodes and then add the ascii parts ; I see no reason why this would be less than the size required, so I think this should be fine.

Reply:
Yeah w can keep this for now and if weird bug showing up this is a suspicious point